app/src/main/java/com/allan/boardbuddies/fragments/TextDialogFragment.java

```diff
@@ -53,18 +48,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     public void onViewCreated(View view, Bundle savedInstanceState) {
         getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);
 
-        editText = view.findViewById(R.id.dialog_input);
-        editText.setText(getArguments().getString("text"));
-        editText.requestFocus();
-        editText.setSelection(editText.getText().length());
+        captionText = view.findViewById(R.id.dialog_input);
+        captionText.setText(getArguments().getString("text"));
+        captionText.requestFocus();
+        captionText.setSelection(captionText.getText().length());
         InputMethodManager imm = (InputMethodManager) requireActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
-        imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT);
+        imm.showSoftInput(captionText, InputMethodManager.SHOW_IMPLICIT);
 
         Button button = view.findViewById(R.id.dialog_button);
         button.setOnClickListener(v -> {
             Bundle result = new Bundle();
-            result.putString("bundleKey", String.valueOf(editText.getText()));
-            getParentFragmentManager().setFragmentResult("requestKey", result);
+            result.putString("TEXT_DIALOG_FRAGMENT_ADDED_TEXT", String.valueOf(captionText.getText()));
+            getParentFragmentManager().setFragmentResult(getArguments().getString("requestKey"), result);
```

Comment:
you renamed the request key so you need to update it here too? otherwise it won't work